Squashed commit of our font/display changes.

s0be opened the pull request:
update display

Beautify display

Add new font generate script, update big font

Signed-off-by: Roman Birg romanbirg@gmail.com

genfont: imrpove parsing

Previously if output ended in the middle of a line (namely bracket was
not in its own line), we'd end up cutting out the last line.

Signed-off-by: Roman Birg romanbirg@gmail.com

genfont: generate whole font file by default

Signed-off-by: Roman Birg romanbirg@gmail.com

Font test

Implment dumb kerning and make all the fonts large

Update font to monospace and better icons

Move fonts to new_fw, fix some bugs and implement display settings

Font_Data: Kerning is not a pointer

Font_Data: Negative kerning is legal

Lines: Fix width bug

Previously lines would draw every which way they wanted
now we can draw a single pixel line and not affect text
around the line

Small cleanup of divergance from upstream

We don't actually have any chanes in Font.h, and font.c
is never used.

Display: Fixup some pointer mistakes

Discovered compiling with clang.

sebirdman committed:
Squashed commit of our font/display changes.

update display

Beautify display

Add new font generate script, update big font

Signed-off-by: Roman Birg <romanbirg@gmail.com>

genfont: imrpove parsing

Previously if output ended in the middle of a line (namely bracket was
not in its own line), we'd end up cutting out the last line.

Signed-off-by: Roman Birg <romanbirg@gmail.com>

genfont: generate whole font file by default

Signed-off-by: Roman Birg <romanbirg@gmail.com>

Font test

Implment dumb kerning and make all the fonts large

Update font to monospace and better icons

Move fonts to new_fw, fix some bugs and implement display settings

Font_Data: Kerning is not a pointer

Font_Data: Negative kerning is legal

Lines: Fix width bug

Previously lines would draw every which way they wanted
now we can draw a single pixel line and not affect text
around the line

Small cleanup of divergance from upstream

We don't actually have any chanes in Font.h, and font.c
is never used.

Display: Fixup some pointer mistakes

Discovered compiling with clang.

ReservedField commented:
It seems to me that `tools/genfont.sh` doesn't specify the kerning field in the `Font_Info_t` struct. Am I missing something?

ReservedField commented:
Also, in `Display_PutText`, is `curX += font->kerning` before the unknown character skip wanted? i.e. should unknown characters have their kerning counted, too?
I'll take care of giving PutLine a little cleanup, the rest looks good.

Commit:
Kerning updates

Add kerning to genfont.sh

Kerning is set to zero by default, depending on the font
the user should set this to their liking

Don't apply kerning to unknown chars

We don't want unknown chars to change the x coord.

ReservedField commented:
Cleaned up, squashed and pushed as 09b100c.
